graphblas/matrix.py

```diff
@@ -791,6 +791,52 @@ def apply(self, op, right=None, *, left=None):
             bt=self._is_transposed,
         )
 
+    def select(self, op, thunk=None):
+        """
+        GrB_Matrix_select
+        Compute SelectOp at each element of the calling Matrix, keeping
+        elements which return True.
+        """
+        if isinstance(op, str):
+            op = select.from_string(op)
+        method_name = "select"
+        if thunk is None:
+            thunk = False  # most basic form of 0 when unifying dtypes
+        if type(thunk) is not Scalar:
+            dtype = self.dtype if (self.dtype._is_udt and not op.is_positional) else None
+            try:
+                thunk = Scalar.from_value(thunk, dtype, is_cscalar=None, name="")
+            except TypeError:
+                thunk = self._expect_type(
+                    thunk,
+                    Scalar,
+                    within=method_name,
+                    keyword_name="thunk",
+                    extra_message="Literal scalars also accepted.",
+                    op=op,
+                )
+        op = get_typed_op(op, self.dtype, thunk.dtype, is_right_scalar=True, kind="select")
+        self._expect_op(op, ("SelectOp", "IndexUnaryOp"), within=method_name, argname="op")
+        if thunk._is_cscalar:
+            if thunk.dtype._is_udt:
+                dtype_name = "UDT"
+                thunk = _Pointer(thunk)
+            else:
+                dtype_name = thunk.dtype.name
+            cfunc_name = f"GrB_Matrix_select_{dtype_name}"
+        else:
+            cfunc_name = "GrB_Matrix_select_Scalar"
+        return MatrixExpression(
+            method_name,
+            cfunc_name,
+            [self, thunk],
+            op=op,
+            expr_repr="{0.name}.select({op}, thunk={1._expr_name})",
+            nrows=self._nrows,
+            ncols=self._ncols,
+            dtype=self.dtype,
+        )
```

Comment:
This doesn't yet work on transposed matrices.  Need to add `at=self._is_transposed` to the MatrixExpression.  Also, should probably add a test to exercise this.

Reply:
Good catch